poollet/bucketpoollet/cmd/bucketpoollet/app/app.go

```diff
@@ -0,0 +1,267 @@
+	fs.DurationVar(&o.DialTimeout, "dial-timeout", 1*time.Second, "Timeout for dialing to the bucket runtime endpoint.")
+	fs.DurationVar(&o.BucketRuntimeSocketDiscoveryTimeout, "bucket-runtime-discovery-timeout", 20*time.Second, "Timeout for discovering the bucket runtime socket.")
+	fs.DurationVar(&o.BucketClassMapperSyncTimeout, "bcm-sync-timeout", 10*time.Second, "Timeout waiting for the bucket class mapper to sync.")
+
+	fs.StringVar(&o.WatchFilterValue, "watch-filter", "", "Value to filter for while watching.")
+}
+
+func (o *Options) MarkFlagsRequired(cmd *cobra.Command) {
+	_ = cmd.MarkFlagRequired("bucket-pool-name")
+	_ = cmd.MarkFlagRequired("provider-id")
+}
+
+func Command() *cobra.Command {
+	var (
+		zapOpts = zap.Options{Development: true}
+		opts    Options
+	)
+
+	cmd := &cobra.Command{
+		Use: "bucketpoollet",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			logger := zap.New(zap.UseFlagOptions(&zapOpts))
+			ctrl.SetLogger(logger)
+			cmd.SetContext(ctrl.LoggerInto(cmd.Context(), ctrl.Log))
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			return Run(ctx, opts)
+		},
+	}
+
+	goFlags := goflag.NewFlagSet("", 0)
+	zapOpts.BindFlags(goFlags)
+	cmd.PersistentFlags().AddGoFlagSet(goFlags)
+
+	opts.AddFlags(cmd.Flags())
+	opts.MarkFlagsRequired(cmd)
+
+	return cmd
+}
+
+func Run(ctx context.Context, opts Options) error {
+	logger := ctrl.LoggerFrom(ctx)
+	setupLog := ctrl.Log.WithName("setup")
+
+	endpoint, err := oriremotebucket.GetAddressWithTimeout(opts.BucketRuntimeSocketDiscoveryTimeout, opts.BucketRuntimeEndpoint)
+	if err != nil {
+		return fmt.Errorf("error detecting bucket runtime endpoint: %w", err)
+	}
+
+	conn, err := grpc.Dial(endpoint,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	if err != nil {
+		return fmt.Errorf("error dialing: %w", err)
+	}
+	defer func() {
+		if err := conn.Close(); err != nil {
+			setupLog.Error(err, "Error closing bucket runtime connection")
+		}
+	}()
+
+	bucketRuntime := ori.NewBucketRuntimeClient(conn)
+
+	cfg, err := configutils.GetConfig(
+		configutils.Kubeconfig(opts.Kubeconfig),
+		configutils.EgressSelectorConfig(opts.EgressSelectorConfig),
+		configutils.WithEgressSelectionName(configutils.EgressSelectionNameControlPlane),
+	)
+	if err != nil {
+		return fmt.Errorf("error getting config: %w", err)
+	}
+
+	leaderElectionCfg, err := configutils.GetConfig(
+		configutils.Kubeconfig(opts.LeaderElectionKubeconfig),
+	)
+	if err != nil {
+		return fmt.Errorf("error creating leader election kubeconfig: %w", err)
+	}
+
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Logger:                  logger,
+		Scheme:                  scheme,
+		MetricsBindAddress:      opts.MetricsAddr,
+		Port:                    9443,
+		HealthProbeBindAddress:  opts.ProbeAddr,
+		LeaderElection:          opts.EnableLeaderElection,
+		LeaderElectionID:        "dfffbeaa.api.onmetal.de",
```

Comment:
This is the same leader election id as for the `volumepoollet` - we should have (a random) other one